src/main.cpp

```diff
@@ -453,6 +454,52 @@ class bearer_token_eviction_manager
     } // evict
 }; // class bearer_token_eviction_manager
 
+class oidc_state_eviction_manager
+{
+    net::steady_timer timer_;
+    std::chrono::seconds interval_;
+
+  public:
+    oidc_state_eviction_manager(net::io_context& _io, std::chrono::seconds _eviction_check_interval)
+        : timer_{_io}
+        , interval_{_eviction_check_interval}
+    {
+        evict();
+    } // constructor
+
+  private:
+    auto evict() -> void
+    {
+        timer_.expires_after(interval_);
+        timer_.async_wait([this](const auto& _ec) {
+            if (_ec) {
+                return;
+            }
+
+            log::trace("Evicting expired OIDC states...");
+            irods::http::process_stash::erase_if([](const auto& _k, const auto& _v) {
+			  const auto* expire_time{boost::any_cast<const std::chrono::steady_clock::time_point>(&_v)};
+
+			  // Not of the right type...
+			  if (expire_time == nullptr) {
+				return false;
+			  }
+
+			  const auto now{std::chrono::steady_clock::now()};
+			  const auto erase_token{now >= *expire_time};
+
+			  if (erase_token) {
+				log::debug("[{}] Evicted state=[{}].", boost::core::demangle(typeid(*this).name()), _k);
```

Comment:
Fancy/different logging style, should probably change to match others.

Reply:
Tell me about this?

Reply:
It's very cool, `typeid(*this).name()` gets the class name, which _might_ be mangled. `boost::core::demangle(...)` ensures we just get the class name, not a mangled version.

The full log line looks like the following:
`[2023-10-31 01:18:54.469] [P:6997] [debug] [T:7009] [oidc_state_eviction_manager] Evicted state=[5ee5b649-a9e7-46d7-ab82-7853d6b53325].`

Reply:
That is nice. We could also hard-code the string to avoid the work to generate that name altogether.

Reply:
True, but then we need to remember to update strings & classes together.
More to do if we plan to spread this out to other classes/things too.

Reply:
Which strings/classes would need to be updated?
Where would we use that outside of this location?

Reply:
Don't think I quite follow.
Were we planning on using this for classes in logging now?

If yes, then, if we hard-code a string in a class while logging, we need to keep that string updated for every class (now and in the future), especially if we ever copy/paste code or update the class name.

The benefit of `typeid(...).name()` is that we avoid having to keep track of these class name strings, with a presumed low cost.

Either way, if we're not going to use this style(?) of logging, it doesn't make sense to keep it here.
Might be of interest in the future though, if we care about logging details in a class.

Reply:
Removed line, since merged with other class.

Reply:
> Were we planning on using this for classes in logging now?

What do you have in mind?

> If yes, then, if we hard-code a string in a class while logging, we need to keep that string updated for every class (now and in the future), especially if we ever copy/paste code or update the class name.

Yes, a valid concern. However, for this specific case, I don't see that happening. The alternative is to do the demangling in the constructor and store it for later use.

> Either way, if we're not going to use this style(?) of logging, it doesn't make sense to keep it here.
Might be of interest in the future though, if we care about logging details in a class.

I agree.